config/environments/test.rb

```diff
@@ -39,13 +35,26 @@
   # Store uploaded files on the local file system in a temporary directory.
   config.active_storage.service = :test
 
+  # Disable caching for Action Mailer templates even if Action Controller
+  # caching is enabled.
   config.action_mailer.perform_caching = false
 
+  # Set the ActiveJob queue adapter for tests
+  # NOTE: the test adapter does *not* automatically perform jobs. To perform jobs,
+  #       call `perform_enqueued_jobs` from the ActiveJob TestHelper at the point
+  #       you want them run. See `spec/rails_helper.rb` for an example of how to
+  #       enable this helper for a class of tests (e.g., feature specs).
+  # config.active_job.queue_adapter = :test
```

Comment:
I added this explicitly. I found that using the `:test` adapter (now default) caused failures with some feature specs and the `:async` adapter caused failures with some request specs. This is because some feature specs need queued jobs to actually run and the `:test` adapter does not run them by default. The `:async` adapter does. However, some request specs check that a job has been enqueued, which the `:async` adapter does not support; only the `:test` adapter does.

Using the `:test` adapter in the test env and then manually performing jobs (via `perform_enqueued_jobs` which comes from `ActiveJob::TestHelper`) where needed appears to be the Rails happy path, so that's what you'll see in this PR.

I don't know how we didn't see this with the prior version of Rails. 🤷🏻‍♂️